src/include/cpu/rpp_cpu_simd.hpp

```diff
@@ -2230,6 +2348,18 @@ inline void rpp_generic_nn_load_i8pln1(Rpp8s *srcPtrChannel, Rpp32s *srcLoc, Rpp
     p = _mm_unpacklo_epi8(px[0], px[1]);    // unpack to obtain [R01|R11|R21|R31|00|00|00|00|00|00|00|00|00|00|00|00]
 }
 
+inline void rpp_generic_nn_load_i8pln1_avx(Rpp8s *srcPtrChannel, Rpp32s *srcLoc, Rpp32s *invalidLoad, __m256i &p)
+{
+    Rpp8s buffer[16] = {0};
+    for(int i = 0; i < 8; i++)
+    {
+        if(!invalidLoad[i])
```

Comment:
Hi Rajy,
I will try this out and get back to you on this

Reply:
Hi Rajy,
Here based on the mask value, we need to load from srcLoc[i] for a given pixel and this can vary for each pixel
_mm256_mask_i32gather_ps does precisely the same operation, but the counterpart is not available for U8 and I8 datatype

I have tried using _mm256_mask_i32gather_epi32 + permute to rearrange the data in required order for U8, but it is degrading the performance. so i have retained the existing code without making any changes